src/IdentityManager/DataService/IdentityServices/CustomPersistedGrantStore.cs

```diff
@@ -0,0 +1,154 @@
+
+            //Logger.LogDebug("removing {persistedGrantCount} persisted grants from database for {@filter}", persistedGrants.Length, filter);
+
+            
+
+            foreach (var item in persistedGrants)
+            {
+                _persistGrantRepository.Remove(item);
+            }
+
+            try
+            {
+                await _unitOfWork.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException ex)
+            {
+                //Logger.LogInformation("removing {persistedGrantCount} persisted grants from database for subject {@filter}: {error}", persistedGrants.Length, filter, ex.Message);
+            }
+        }
+
+        public async Task RemoveAsync(string key)
+        {
+            var persistedGrant = (await _persistGrantRepository.GetAll().Where(x => x.Key == key).ToArrayAsync())
+               .SingleOrDefault(x => x.Key == key);
+            if (persistedGrant != null)
+            {
+                //Logger.LogDebug("removing {persistedGrantKey} persisted grant from database", key);
+
+                _persistGrantRepository.Remove(persistedGrant);
+
+                try
+                {
+                    await _unitOfWork.SaveChangesAsync();
+                }
+                catch (DbUpdateConcurrencyException ex)
+                {
+                    //Logger.LogInformation("exception removing {persistedGrantKey} persisted grant from database: {error}", key, ex.Message);
+                }
+            }
+            else
+            {
+                //Logger.LogDebug("no {persistedGrantKey} persisted grant found in database", key);
+            }
+        }
+
+        public async Task StoreAsync(PersistedGrant grant)
+        {
+            var existing = (await _persistGrantRepository.GetAll().Where(x => x.Key == grant.Key).ToArrayAsync())
+                .SingleOrDefault(x => x.Key == grant.Key);
+            if (existing == null)
+            {
+                //Logger.LogDebug("{persistedGrantKey} not found in database", token.Key);
+
+                var persistedGrant = grant.ToEntity();// map to db model
+                _persistGrantRepository.Add(persistedGrant);
+            }
+            else
+            {
+                //Logger.LogDebug("{persistedGrantKey} found in database", token.Key);
+
+
+
+                grant.UpdateEntity(existing);
+            }
+
+            try
+            {
+                await _unitOfWork.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException ex)
+            {
+                //Logger.LogWarning("exception updating {persistedGrantKey} persisted grant in database: {error}", token.Key, ex.Message);
+            }
+        }
+
+        private IQueryable<Travely.IdentityManager.Repository.Abstractions.Entities.PersistedGrant> Filter(IQueryable<Travely.IdentityManager.Repository.Abstractions.Entities.PersistedGrant> query, PersistedGrantFilter filter)
```

Comment:
You can use c# using aliasies, to avoid looooong namespaces and same name issue
Example `using PersistedGrantEntity = Travely.IdentityManager.Repository.Abstractions.Entities.PersistedGrant;`